core/src/main/java/org/mini2dx/connect4/Board.java

```diff
@@ -124,14 +132,206 @@ private void checkInput() {
+                addTileAt(bTile.clone(), selectedCol);
                 isBluesTurn = false;
+                selectionTile.setColour(RED);
         } else {
-                addTileAt(rTile.clone(), selectCol);
+                addTileAt(rTile.clone(), selectedCol);
                 isBluesTurn = true;
+            selectionTile.setColour(BLUE);
+        }
+    }
+
+    private boolean hasGameBeenWon(int selectedRow)
+    {
+
+        return verticalWin() || horizontalWin(selectedRow) || diagonalWin(selectedRow);
+
+    }
+
+    private boolean verticalWin()
+    {
+        int colourMatchs = 0, colourMismatch = 0;
+        int row = 5;
+        Tile.TileColour colour = board[row][selectedCol].getColour();
+
+        do
+        {
+            if(colour.equals(board[row][selectedCol].getColour()))
+            {
+                colourMatchs++;
+                 if(colourMatchs == WIN_CONDITION)
+                     return true;
+
+            }
+
+            else
+            {
+                colourMismatch++;
+                colourMatchs = 0;
+                if(colourMismatch > (BOARD_SIZE - WIN_CONDITION))
+                    return false;
+
+            }
+
+            row--;
+        }
+        while(row >= 0);
+
+        return false;
+    }
+
+    private boolean horizontalWin(int selectedRow)
+    {
+        int colourMatches = 0, colourMismatches = 0;
+        Tile.TileColour colour = board[selectedRow][selectedCol].getColour();
+
+        for(int col = 0; col < BOARD_SIZE; col++)
+        {
+            if(colour.equals(board[selectedRow][col].getColour()))
+            {
+                colourMatches++;
+                if(colourMatches == WIN_CONDITION)
+                    return true;
+            }
+
+            else
+            {
+                colourMismatches++;
+                colourMatches = 0;
+                if(colourMismatches > (BOARD_SIZE - WIN_CONDITION))
+                    return false;
+
+            }
+        }
+        return false;
+    }
+
+    private boolean diagonalWin(int selectedRow)
+    {
+        return bottomLeftDiagonal(selectedRow) || bottomRightDiagonal(selectedRow) || topLeftDiagonal(selectedRow) ||
+                topRightDiagonal(selectedRow);
+    }
+
+    public boolean bottomLeftDiagonal(int selectedRow)
+    {
+        int colourMatches = 0, colourMismatches = 0;
+        Tile.TileColour colour = board[selectedRow][selectedCol].getColour();
+        int stepX = selectedRow;
+        int stepY = selectedCol;
+
+        while(stepX < BOARD_SIZE && stepY < BOARD_SIZE)
+        {
+            if(colour.equals(board[stepX][stepY].getColour()))
+            {
+                colourMatches++;
+                if(colourMatches == WIN_CONDITION)
+                    return  true;
+            }
+            else
+            {
+                colourMismatches++;
+                colourMatches = 0;
+                if(colourMismatches > (BOARD_SIZE - WIN_CONDITION))
```

Comment:
Use brackets around all if code paths even if it's only 1 line. Improves readability.